Modules/zipimport.c

```diff
@@ -651,7 +651,8 @@ zipimport_zipimporter_get_data_impl(ZipImporter *self, PyObject *path)
     Py_ssize_t path_start, path_len, len;
 
 #ifdef ALTSEP
-    path = _PyObject_CallMethodId(path, &PyId_replace, "CC", ALTSEP, SEP);
+    path = _PyObject_CallMethodId((PyObject *)&PyUnicode_Type, &PyId_replace,
+                                  "OCC", path, ALTSEP, SEP);
```

Comment:
I am not sure about this.
would it be better to simply check whether the return value of pathname.replace('/', '\\') is a str?

Reply:
The current solution LGTM.